utils/script.c

```diff
@@ -124,8 +125,10 @@ int script_init(struct script_info *info, enum uftrace_pattern_type ptype)
 		script_pathname = NULL;
 	}
 
-	if (script_pathname == NULL)
+	if (script_pathname == NULL) {
+		script_finish_filter();
```

Comment:
Hmm.. I still don't like this.  I guess it might leak some resources like `python_handle` and many memory objects in the cpython internal when module loading is failed..